assemble/build.sh

```diff
@@ -15,134 +15,266 @@
+
+  echo
+  local ignore
+  read -s -p 'Press Enter to continue or Ctrl+c to exit...' ignore
+  echo 1>&2
+
+  local cBranch
+  cBranch=$(currentBranch) || fail $(red Failure) to get current branch from git
+  cacheGPG || fail "Unable to cache GPG credentials into gpg-agent"
+
+  local rc; rc=$(prompter 'release candidate number (eg. 1, 2, etc.)' '[0-9]+')
+  local rcBranch; rcBranch=$ver-rc$rc
+  local nBranch; nBranch=$cBranch-$rcBranch-next
+
+  # create working branch
+  run git branch $nBranch $cBranch && \
+    run git checkout $nBranch ||
+    fail "Unable to create working branch $(red $nBranch) from $(red $cBranch)!"
+
   # create a release candidate from a branch
-  run mvn clean release:clean release:prepare release:perform
+  local oFile; oFile=/tmp/accumulo-build-$rcBranch.log
+  rm -f $oFile && \
+  runLog $oFile mvn clean release:clean || \
+    cleanUpAndFail 'mvn clean release:clean' $oFile $cBranch $nBranch
+  runLog $oFile mvn -B release:prepare || \
+    cleanUpAndFail 'mvn release:prepare' $oFile $cBranch $nBranch
+  runLog $oFile mvn release:perform || \
+    cleanUpAndFail 'mvn release:perform' $oFile $cBranch $nBranch
+
+  # verify the next branch contains both expected log messages and no more
+  [[ $(gitCommits ${cBranch}..${nBranch} | wc -l) -eq 2 ]] && \
+  [[ $(gitCommit  ${nBranch}~2) =  $(gitCommit ${cBranch}) ]] && \
+  [[ $(gitSubject ${nBranch})   =~ ^\[maven-release-plugin\]\ prepare\ for\ next ]] && \
+  [[ $(gitSubject ${nBranch}~1) =~ ^\[maven-release-plugin\]\ prepare\ release\ $ver ]] || \
+    cleanUpAndFail "verifying that $nBranch contains only logs from release plugin"
+
+  # verify the tag is one behind $nBranch and one ahead of $cBranch
+  [[ $(gitCommit ${nBranch}~1) = $(gitCommit refs/tags/rel/$ver) ]] || \
+    cleanUpAndFail "verifying that ${nBranch}~1 == refs/tags/rel/$ver"
+
+  # remove tag which was created
+  run git tag -d rel/$ver || \
+    cleanUpAndFail "removing unused git tag rel/$ver"
+
+  # create release candidate branch to vote on
+  run git branch $rcBranch ${nBranch}~1 || \
+    cleanUpAndFail "creating branch $rcBranch"
+
+  # push branches (ask first)
+  local origin; origin=$(git remote -v | grep ^origin | grep push | awk '{print $2}')
+  echo "Do you wish to push the following branches to origin ($(yellow $origin))?"
+  echo "  $nBranch"
+  echo "  $rcBranch"
+  local a; a=$(prompter "letter 'y' or 'n'" '[yn]')
+  [[ $a == 'y' ]] && \
+    run git push origin refs/heads/$nBranch refs/heads/$rcBranch || \
+    red "Did not push branches; you'll need to perform this step manually."
+
+  # continue to creating email notification
+  local stagingrepo
+  stagingrepo=$(<$oFile grep -o 'repository[.]apache[.]org/content/repositories/orgapacheaccumulo-[0-9]*' | tail -1)
+  stagingrepo=${stagingrepo##*-}
+  echo $(red Running) $(yellow $scriptname --create-email $stagingrepo $ver $rc)
+  createEmail $stagingrepo $ver $rc
+}
+
+if [[ $1 = '--create-release-candidate' ]]; then
```

Comment:
There's lots of other prerequisites (depending on branch being built), and assumptions we could avoid, but I tried to use standard tools to simplify things a bit, rather than check for everything. I leave some things up to the user to configure their own environment properly.

I tried to support both automatically, but it was a bit problematic to do that. gpg2 has been around for awhile now, and gpg1 is considered legacy or "classic", though it's still supported. Most modern distributions include gpg2, and it was necessary to use gpg2 to get the gpg-agent integration with Gnome Shell (rather than require launching a separate gpg-agent).